specs/mgt-search-suggestion.md

```diff
@@ -0,0 +1,161 @@
+# mgt-search-suggestion
+
+The suggestion component based on Graph Suggestion API, it provides a input box and flyout. When a query string typed in the input box, the flyout will rendered by the data from Graph Suggestion API, show some suggestions. it supports 3 kinds of entity types, File/Text/Suggestions. The component structure as below
+<img src="./images/mgt-search-suggestion.png" width=400/>
+
+## Supported functionality
+
+| Feature | Priority | Notes |
+| ------- | -------- | ----- |
+| Retreieve text/people/file/ from Microsoft Graph endpoint based on the query string | P0 | |
+| Display the text of the matched query and display the name of the file/people| P0| |
+| Display the headImg or create a head icon for suggested people| P0| |
+| Display an icon indicating if it's a folder or file and the file type| P0 | Icons needed include generic folder icon, .docx, .pptx, .xlsx, and generic file icon for other file types |
+| Display relevant details of the file | P0 | Developer should be able to configure what details are being rendered |
+| Provide basic default key down action and mouse action | P0 | when mouse move to an item, the backgroud/cursor should change. when arrow up/down, one item should be selected|
+| Provide callback function to get data for supporting users customization actions | P0 | when user click one item, should provide a way for getting the suggested data and support customize actions. when enter key pressed, should provide a way for getting the focused data and support customize actions|
+
+## Templates
+
+ `mgt-search-suggestion` supports several [templates](../customize-components/templates.md) that you can use to replace certain parts of the component. To specify a template, include a `<template>` element inside a component and set the `data-type` value to one of the following.
+
+| Data type | Data context | Description |
+| --- | --- | --- |
+| default | null: no data | The template used to override the rendering of the entire component.
+| loading | null: no data | The template used to render the state of picker while request to graph is being made. |
```

Comment:
We should be clear what part of the component is templated with the `loading` template. There are two I can think of
- the entire component when the component is initialized - this is only if the component needs to make any calls to graph before it can be used - not sure if super useful for a picker
- the contents of the suggestions dropdown, while it is searching - this seems more relevant as the user can replace a spinner with a custom progress indicator

Reply:
yeah, for this component, I don't think I need initialized the list, I will set a check, when user type something, there will have loading action. 
How do you think about it?

Reply:
I think that makes